nixos/modules/security/systemd-confinement.nix

```
@@ -160,7 +160,8 @@ in {
               + " the 'users.users' option instead as this combination is"
               + " currently not supported.";
     }
-    { assertion = !cfg.serviceConfig.ProtectSystem or false;
+    { assertion = ! cfg.serviceConfig ? ProtectSystem ||
+                  (lib.isBool cfg.serviceConfig.ProtectSystem && !cfg.serviceConfig.ProtectSystem);
```

I know I'm being somewhat nitpicky, but I find this hard to read, what about this instead?

```suggestion
    { assertion = cfg.serviceConfig ? ProtectSystem -> cfg.serviceConfig.ProtectSystem == false;
```

---

Sure, I didn't know that syntax. I don't understand how it helps with type incompatibility (option is set as a string and compared to a bool), but it appears to works so accepted it.

For my culture, I read this as "if cfg.serviceConfig has ProtectSystem attribute then it also has to be equal to false", is that right? In that case why doesn't it fail with string comparison?

---

> For my culture, I read this as "if cfg.serviceConfig has ProtectSystem attribute then it also has to be equal to false", is that right?

That's correct.

> In that case why doesn't it fail with string comparison?

The reason is that the `assignment` attribute is expected to be of type boolean, eg. if you do something like `if "foo" then ...` it will fail. However, since we're comparing two **different** types here, equality is *always* `false`, like eg. `"foo" == false`.